### PR TITLE
Refactor: RabbitHole is made a component of CheshireCat to allow auto-ingest

### PR DESCRIPTION
**Pull request by nicola-corbellini**
The RabbitHole is now a class embedding the function "ingest_file".
CheshireCat class creates one RabbitHole instance at initialization time and can call "ingest_file".
Upload.py has been updated to make the ingestion works both from GUI and scripts

### CODE REVIEW

**Comment by pieroit** [comment]
Great jobs thanks!
There are a few things to fix before merge, please see comments in the PR

File: web/cat/looking_glass/cheshire_cat.py [hunk 1]
```diff
@@ -18,6 +21,9 @@ def __init__(self, verbose=True):
         # bootstrap the cat!
         self.bootstrap()
 
+        # Rabbit Hole Instance
+        self.rb_hole = RabbitHole()
```
Comment (pieroit) [inline web/cat/looking_glass/cheshire_cat.py:25]: please call the attribute `self.rabbit_hole` as it is more readable

File: web/cat/routes/upload.py [hunk 2]
```diff
@@ -15,11 +14,6 @@ async def rabbithole_upload_endpoint(
     request: Request,
     file: UploadFile,
     background_tasks: BackgroundTasks,
-    chunk_size: int = Body(
```
Comment (pieroit) [inline web/cat/routes/upload.py:18]: Add back `chunk_size` and `chunk_overlap` params to the endpoint, so scripts uploading files to the Cat can decide how the text is splitted and overlapped

File: web/cat/routes/upload.py [hunk 3]
```diff
@@ -42,9 +36,7 @@ async def rabbithole_upload_endpoint(
         )
 
     # upload file to long term memory, in the background
-    background_tasks.add_task(
-        ingest_file, ccat, file, chunk_size=chunk_size, chunk_overlap=chunk_overlap
-    )
+    background_tasks.add_task(ccat.send_file_in_rabbit_hole, file)
```
Comment (pieroit) [inline web/cat/routes/upload.py:39]: the function call should take `chunk_size` and `chunk_overlap` args as it was before

Comment (pieroit) [reply web/cat/routes/upload.py:39]: @nicola-corbellini to try out the endpoints see `localhost:1865/docs/`

**Review by pieroit — CHANGES_REQUESTED** [review]
Can you fix this stuff?

File: web/cat/routes/upload.py (hunk 3 above)
Comment (nicola-corbellini) [reply web/cat/routes/upload.py:39]: Ouch sorry, I naively misunderstood the code. 
I'll find a way to fix this, as before removing the `chunk_size` and `chunk_overlap` params from the endpoint it raised an error. I'll dig into this!

**Comment by pieroit** [comment]
@nicola-corbellini congrats on your first successful PR!
Many thanks :)

**Pull request merged**